lms/djangoapps/grades/tasks.py

```diff
@@ -34,6 +40,26 @@ def recalculate_subsection_grade(user_id, course_id, usage_id, only_if_higher):
     course_key = CourseLocator.from_string(course_id)
     student = User.objects.get(id=user_id)
     scored_block_usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
+    if modified_time_string:
+        student_module = StudentModule.objects.get(
+            student_id=user_id,
+            module_state_key=scored_block_usage_key,
+            course_id=course_key,
+        )
+        read_time = student_module.modified
+        modified_time = datetime.strptime(modified_time_string, "%y/%m/%d/%H/%M/%S/%f")
+        modified_time = timezone.make_aware(modified_time, timezone=None)
+        log.warning("read_time: {}, modified_time:{}".format(read_time, modified_time))
+        if read_time > modified_time:
```

Comment:
Please add a comment here of the rationale for this check.  I'm sure future readers will wonder why.